Source/DafnyDriver/DafnyDriver.cs

```diff
@@ -283,17 +287,19 @@ public enum CommandLineArgumentsResult {
           }
           dafnyFiles.Add(df);
           isDafnyFile = true;
+        } catch (ArgumentException e) {
+          options.Printer.ErrorWriteLine(options.ErrorWriter, "*** Error: {0}: ", nameToShow, e.Message);
+          return CommandLineArgumentsResult.PREPROCESSING_ERROR;
         } catch (IllegalDafnyFile e) {
           if (e.ProcessingError) {
             return CommandLineArgumentsResult.PREPROCESSING_ERROR;
           }
           // Fall through and try to handle the file as an "other file"
+        } catch (Exception e) {
+          options.Printer.ErrorWriteLine(options.ErrorWriter, "*** Error: {0}: {1}", nameToShow, e.Message);
+          return CommandLineArgumentsResult.PREPROCESSING_ERROR;
```

Comment:
For the record I generally don't like losing the stack trace for unexpected exceptions, but I appreciate that this code is already a bit of a mess w.r.t. error messages so I'm not going to block on it. I've especially never been a fan of the only-used-for-control-flow exception `IllegalDafnyFile` :)